### DISCUSSION TITLE
Testing default values for query parameters

### DISCUSSION THREAD

**Discussion by oren0e** [rwf2/Rocket#2133] [Questions]
Hi,
If I have an endpoint like:
```rust
#[get("/known-issues?<id>&<only_first>", rank = 1)]
pub fn find_known_issues(
...
```
Is there a way that I can access the request's query parameter values that were sent? Basically I want to test that if I don't pass the `only_first` flag (which is a boolean) then it defaults to false.

**Comment by somehowchris** [comment]
You could either use a `Option` type and then use `unwrap_or_default` or use a struct for your query params which then uses the `FromForm` style with which you can set a default https://rocket.rs/v0.5-rc/guide/requests/#defaults

**Reply by somehowchris** [reply]
If you like to make it fun and complicated, create your own data guard to set defaults on type `T`

**Reply by oren0e** [reply]
@somehowchris wait, if I hit that endpoint without the `only_first` parameter then in the logs I see it as False, I assumed this is some default behaviour of rocket, and if it is - then how can I access that "False"? Utilizing the existing rocket solutions

**Reply by oren0e** [reply]
And I also don't use forms in my code, it's just a rest endpoint

**Comment by oren0e** [comment]
I saw that it is possible to do:
```rust
    let req = client
        .get(format!("/api/known-issues?id={}", encoded_id));
    println!("{:?}", req);
    let inner_req = req.inner();
```
and then `inner_req` can get me the query param by using `.get_query_value()` - but the problem is that the resulting request struct is pretty much empty:
```
Request { method: Cell { value: Get }, uri: Origin { source: None, path: "/", query: None, segment_count: [uninitialized
 storage] }, headers: HeaderMap { headers: {} }, remote: None }
```
How is it possible? Why is it empty despite the fact that I pass query params? (and presumably the missing `only_first` param should be there too with its default value, as I would expect)
